CorsixTH/Lua/dialogs/resizables/file_browser.lua

```diff
@@ -197,23 +197,40 @@ function UIFileBrowser:UIFileBrowser(ui, mode, title, vertical_size, root, show_
   -- Initialize the tree control
   self.control = FilteredTreeControl(root, 5, 35, h_size - 10, vertical_size, self.col_bg, self.col_scrollbar, true, show_dates)
     :setSelectCallback(--[[persistable:file_browser_select_callback]] function(node)
-      if node.is_valid_file and (lfs.attributes(node.path, "mode") ~= "directory") then
-        self:choiceMade(node.path)
-      end
+      if self:checkChoice(node) then self:choiceMade(node.path) end
+    end)
+    :setTextboxCallback(--[[persistable:file_browser_textbox_callback]] function(node, label)
+      if self.updateTextbox and self:checkChoice(node) then self:updateTextbox(label) end
```

Comment:
Another way to remove the first check is probably making an empty `updateTextbox` in `UIFileBrowser` that just returns false. But maybe this way is better?

Reply:
Just make the method always available so you can call it without checking first by adding `function UIFileBrowser:updateTextbox(label) error("Override me"); end` The derived classes also implement that method so at runtime the latter will be executed. `UIFileBrowser:updateTextbox()` is a placeholder, saying "This will be called for updating the textbox, override it for a proper implementation in the derived class."

Same with `confirmName` below as far as I can see. Perhaps there are more (didn't check).

If you think in classes and derived classes, it's normal that the base class provides hooks for use by the derived classes in the form of empty or basic methods that are intended to be changed.
It's the standard way to customize behavior of a class by inserting custom code in base class code.

The `error()` call in my example forces that a derived class must provide a new function for it. If you don't want to force overriding, eg when skipping the update is fine too, you can provide an empty function in the base class.  
The derived class then has the option (but not the obligation) to provide a new implementation.

Overriding methods is mostly the same as our anonymous `(function() blah blah end)` things, except classes give more structured use to such functions, as it always has a well-defined class+method name.